platform/wab/src/wab/client/components/sidebar/LeftGeneralTokensPanel.tsx

```diff
@@ -364,20 +353,46 @@ const LeftGeneralTokensPanel = observer(function LeftGeneralTokensPanel() {
           }}
         >
           <VirtualTree
-            ref={treeRef}
             rootNodes={items}
-            getNodeKey={getRowKey}
-            getNodeChildren={getRowChildren}
-            getNodeSearchText={getRowSearchText}
-            getNodeHeight={getRowHeight}
-            query={debouncedQuery}
             renderElement={TokenTreeRow}
+            nodeData={nodeData}
+            nodeKey={nodeKey}
+            nodeHeights={nodeHeights}
+            expandAll={expandAll}
+            collapseAll={collapseAll}
           />
         </TokenControlsContext.Provider>
       </MultiAssetsActions>
     );
   };
 
+  const treeItems = React.useMemo(
```

Comment:
I had to extract this from `tokenSectionItems` to provide to `useTreeData` so that `expandAll` and `collapseAll` are available at the outer level. I verified everything that it didn't break anything as best I could, but it seems like there are probably a few more simplifications that could be made. Might be a bit of a time sink though.